### PR TITLE
Read credentials from mobile-services.json

### PR DESCRIPTION
**Pull request by danielpassos**
Related issues:

* [AGIOS-680](https://issues.jboss.org/browse/AGIOS-680)

### CODE REVIEW

**Comment by david-martin** [comment]
@pb82 how does this align with tasks in https://trello.com/c/eYDEQK1o/43-8-pushconfig-to-mar-sync-as-a-developer-i-want-the-configuration-generated-by-the-ups-server-to-be-available-in-the-associating

File: modules/push/AgsPush.swift [hunk 1]
```diff
@@ -30,8 +31,13 @@ public class AgsPush {
         - mobileConfig: the configuration for the auth service from the service definition file
      */
     private init(_ mobileConfig: MobileService?) {
-        self.serverURL = URL(string: (mobileConfig?.url)!)!
         self.requestApi = AgsCore.instance.getHttp()
+        self.serverURL = URL(string: (mobileConfig?.url)!)!
+
+        let variant = mobileConfig?.config!["ios"]?.getObject()?["variantId"]?.getString()
```
Comment (wtrocki) [inline modules/push/AgsPush.swift]: 1) Is `ios` a good key? Strange that IOS is used as platform is known (for mobile app id).
@david-martin @pb82 

2) To not unwrap optionals many times we could write something like this:

```swift
let pushSecurityConfig = mobileConfig?.config!["ios"]?.getObject();
let variant = pushSecurityConfig?["variantId"]?.getString()
let secret = pushSecurityConfig??["variantSecret"]?.getString()

```

Comment (danielpassos) [reply modules/push/AgsPush.swift]: 1. Well, I've used the same tree of UPS standard alone.

**Comment by wtrocki** [comment]
I have also noticed that unit tests were removed in previous PR
Can we restore push unit tests for registration.

File: example/AeroGearSdkExample/mobile-services.json [hunk 2]
```diff
@@ -23,13 +23,18 @@
...
-   {
-        "id": "push",
-        "name": "push",
-        "type": "push",
-        "url": "https://ups-dm-myproject-1.193b.starter-ca-central-1.openshiftapps.com/",
-        "config": {}
-   }
+    },
+    {
+      "id": "push",
```
Comment (pb82) [inline example/AeroGearSdkExample/mobile-services.json]: @danielpassos @wtrocki I think in the end the config generated by the mobile cli would look more like this:

```
{
  "name": "push",
  "variants": [
    {
        "uri": "https://ups-dm-myproject-1.193b.starter-ca-central-1.openshiftapps.com/",
        "type": "android",
        "googlekey": "...",
         ...
    }
  ]
}
```

Comment (matzew) [reply example/AeroGearSdkExample/mobile-services.json]: does this make it harder to parse the file? since we now have no `ios` or `droid` objects ?

Comment (danielpassos) [reply example/AeroGearSdkExample/mobile-services.json]: For now, all kinds of stuff should be under `config` node. There is no way to get `rootNode.variants`

**Pull request merged**